lib/TeleIrc.js

```diff
@@ -35,10 +35,6 @@ class TeleIrc {
       this.errorCodes = new TeleIrcErrorCodes();
     }
 
-    if (this.config.imgur.useImgurForImageLinks) {
-      imgur.setClientId(this.config.imgur.imgurClientId);
```

Comment:
As far as I know, this call is still needed, it should probably happen in a constructor ....elsewhere.

Reply:
@thenaterhood Any suggestions here? I tested the bot fully and it seemed to work without this code, but I couldn't figure out where it belongs. The problem is that the `setClientId()` method is not used anywhere else in the entire project that I could find. This is why I assumed it was legacy code.

Reply:
If I recall correctly, imgur was a global in Teleirc.js.  It was at the top of the file, and it looked like:

```js
let imgur = require("imgur");
```

However, I must have missed this during the massive refactor.  Right now, the require("imgur") is no longer a global (why create it if the user is not using imgur?).  It is passed into the Imgur Telegram Handler's constructor:

```js
    constructor(imgur, enabled, tgbot, action) {
        super(enabled, tgbot, action);
        this._imgur = imgur;
}
```

Which is invoked in initStage5_initTelegramMessageSending in Teleirc.js:

```js
    if (this.config.imgur.useImgurForImageLinks) {
      this.tgPhotoHandler = new TgImgurPhotoHandler(
        require('imgur'),
        true,
        this.tgbot,
        this.sendMessageToIrc.bind(this)
      );
}
```

I'm not entirely sure what setClientId does, since the library's documentation is non-existent.  It _could_ be like an API key???  Whether or not its required for imgur to work, I'm not sure.

If we *do* need it, we might want to consider adding an additional parameter to TgImgurPhotoHandler that takes in the config's imgur class, and calling setClientId inside of it...  Something like:

```js
    constructor(imgur, imgurConfig, enabled, tgbot, action) {
        super(enabled, tgbot, action);
        this._imgur = imgur;
        this._imgur.setClientId(imgurConfig.imgurClientId);
}
```

Note: The mock imgur class in the unit tests will need an empty setClientId function as well.

Reply:
Just curious, @jwflory, is there a file in ~/.imgur whose contents look like a client ID?  That might explain why its still working... it could be loading it to disk.

Reply:
Actually, nevermind, I looked at imgur's source code, and saw this:

```js
// The following client ID is tied to the
// registered 'node-imgur' app and is available
// here for public, anonymous usage via this node
// module only.
var IMGUR_CLIENT_ID = process.env.IMGUR_CLIENT_ID || 'f0ea04148a54268';
```

It looks like if a client ID is not specified, it uses some default one.  We probably don't want that.

Reply:
@xforever1313 I didn't see a written `imgur.` file in a running instance with the current `HEAD` of the `master` branch.

Reply:
@xforever1313 I see. So we need to set the API token as an environment variable? I assume my success with using Imgur in testing was with the default API key then?

Reply:
@jwflory yep, if you've done this in testing, you would have used the default client id.

In order to upload to imgur, you'll want to send an HTML message. With JS, you'll want to send a JSON POST request.

[Here](https://git.xfce.org/apps/xfce4-screenshooter/tree/lib/screenshooter-imgur.c#n104) is the implementation in C.

[Here](https://ctrlq.org/code/20526-javascript-image-uploader) is information on how to achieve this in JS. 

The general format is looking for an *Authorization* field and appending the Client-ID + API key.

Reply:
@Tjzabel @xforever1313 The code for this seems to be here:

https://github.com/RITlug/teleirc/blob/master/lib/TelegramHandlers/TgImgurPhotoHandler.js#L25-L37

But I can't take a guess of how to do this. I don't understand where the Imgur API URL is being formed or how this handler knows what the URL is.

Reply:
@jwflory @xforever1313 From the code snippet linked, it seems `_UploadImage` is a function override, and `uploadUrl` is the function that does the actual image uploading to Imgur. I can't find any implementation of this function, except in the test cases. 

In the constructor [here](https://github.com/RITlug/teleirc/blob/master/lib/TelegramHandlers/TgImgurPhotoHandler.js#L13), there is an `imgur` object, but I don't know what its purpose is.

It seems we need to create the `uploadUrl` function to handle the actual uploading process within the `TgImgurPhotoHandler.js` file.

Reply:
Moved comment thread over to issue #93